impl/client/src/main/javascript/web/pentaho/lang/EventSource.js

```diff
@@ -533,17 +533,40 @@ define([
     return [type];
   }
 
-  function emitQueue(event, queue, type, phase, isCanceled, keyArgs) {
+  /**
+   * Emits an event.
+   *
+   * @this pentaho.lang.EventSource
+   *
+   * @param {object} event - The event payload object.
+   * @param {!Array.<pentaho.lang.IEventObserverRegistration>} queue - The queue of event observer regitrations.
+   * @param {nonEmptyString} type - The event type.
+   * @param {nonEmptyString} phase - The event phase.
+   * @param {function(object):boolean} [isCanceled] - A predicate that indicates if a given event object
+   * is in a canceled state.
+   * @param {function(any, object, nonEmptyString, nonEmptyString)} [errorHandler] -
+   * When specified with a `null` value,
+   * no error handling is performed and errors thrown by listeners are thrown back to this method's caller.
+   * When unspecified or specified as `undefined`, defaults to a function that simply log the listener errors,
+   * and let execution continue to following listeners.
+   * The function arguments are: the error, the event, the event type and the event phase.
```

Comment:
If we need to document the callback, then we should create a doclet just for that...